polygon/sync/sync.go

```diff
@@ -0,0 +1,210 @@
+package sync
+
+import (
+	"context"
+
+	"github.com/ledgerwatch/log/v3"
+
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/polygon/p2p"
+)
+
+type Sync struct {
+	storage          Storage
+	execution        ExecutionClient
+	verify           AccumulatedHeadersVerifier
+	p2pService       p2p.Service
+	downloader       *HeaderDownloader
```

Comment:
oh I just noticed this - we should maybe have `HeaderDownloader` be an interface and use it as the type here and `headerDownloader` be the struct so that the `Sync` can mock the header downloader in tests if needed

Reply:
👌 could you do this change in one of your PRs?